tests/integration/os.rs

```diff
@@ -91,6 +100,62 @@ async fn return_result_cairo1_account(
     .expect("OS run failed");
 }
 
+#[rstest]
+// We need to use the multi_thread runtime to use task::block_in_place for sync -> async calls.
+#[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+async fn declare_cairo1_account(
+    #[future] initial_state_cairo1: StarknetTestState,
+    block_context: BlockContext,
+    max_fee: Fee,
+) {
+    let mut initial_state = initial_state_cairo1.await;
+
+    let tx_version = TransactionVersion::TWO;
+    let mut nonce_manager = NonceManager::default();
+
+    let account_contract = initial_state.cairo1_contracts.get("account_with_dummy_validate").unwrap();
+
+    // we want to declare a fresh (never-before-declared) contract, so we don't want to reuse
+    // anything from the test fixtures, and we need to do it "by hand"
+    let (_, sierra_class, casm_class) = load_cairo1_contract("empty_contract");
+    let mut ffc: FactFetchingContext<DictStorage, PoseidonHash> =
+        initial_state.cached_state.state.ffc.clone_with_different_hash();
+    let (contract_class_hash, compiled_class_hash) =
+        write_class_facts(sierra_class.clone(), casm_class.clone(), &mut ffc).await.unwrap();
+
+    let sender_address = account_contract.address;
+
+    let contract_class = compiled_contract_class_cl2vm(&casm_class).unwrap();
+    let class_hash = starknet_api::core::ClassHash::try_from(contract_class_hash).unwrap();
+    let compiled_class_hash = CompiledClassHash::try_from(compiled_class_hash).unwrap();
+
+    let class_info = ClassInfo::new(&contract_class, sierra_class.sierra_program.len(), 0).unwrap();
+
+    let declare_tx = blockifier::test_utils::declare::declare_tx(
+        declare_tx_args! {
+            max_fee,
+            sender_address,
+            version: tx_version,
+            nonce: nonce_manager.next(sender_address),
+            class_hash: class_hash.into(),
+            compiled_class_hash,
+        },
+        class_info,
+    );
+
+    initial_state.cairo1_compiled_classes.insert(class_hash, casm_class);
```

Comment:
I'm surprised that you have to modify the initial state here. If I understand clearly, in production today, you need to send the Sierra class entirely and Starknet will compile it to CASM. This is obviously not something we're going to support, so I guess the flow here imitates the result of the compilation step?

I think all state manipulations should be isolated and documented to explain why we're doing this as it's not trivial.

Reply:
Good catch, this isn't necessary, removed in a6dd142